### PR TITLE
feat: added guide for using existing table as vector store

### PR DESCRIPTION
**Pull request by vishwarajanand**
Added guide for re-using an existing table to `VectorStore`

Design Doc: [go/content-metadata-guide-alloydb](http://goto.google.com/content-metadata-guide-alloydb)

### CODE REVIEW

File: docs/vector_store.ipynb [hunk 1]
```diff
@@ -563,6 +560,115 @@
...
+    "    description TEXT,\n",
+    "    price_usd DECIMAL(10, 2) NOT NULL,\n",
+    "    category VARCHAR(255),\n",
+    "    quantity INT DEFAULT 0,\n",
+    "    sku VARCHAR(255) UNIQUE NOT NULL,\n",
+    "    embed vector(768) DEFAULT NULL, --> vector dimensions depends on the embedding model\n",
+    "    image_url VARCHAR(255)\n",
+    ");\n",
+    "\n",
+    "-- Insertion of records\n",
```
Comment (averikitsch) [inline docs/vector_store.ipynb:590]: Let's not highlight the use case of inserting NULL embeddings. See comment on fully working example

File: docs/vector_store.ipynb [hunk 2]
```diff
@@ -503,7 +503,8 @@
     "## Create a custom Vector Store\n",
     "A Vector Store can take advantage of relational data to filter similarity searches.\n",
     "\n",
-    "Create a table with custom metadata columns."
+    "Create a new table with custom metadata columns.\n",
+    "You can also re-use an existing table which already has custom columns for content, id, embedding and/or metadata."
```
[the remark this answers is not in the document]
Comment (vishwarajanand) [reply docs/vector_store.ipynb]: Added

File: docs/vector_store.ipynb [hunk 3]
```diff
@@ -563,6 +560,115 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import uuid\n",
+    "\n",
+    "# Use filter on search\n",
+    "docs = await custom_store.asimilarity_search_by_vector(\n",
+    "    query_vector, filter=\"price_usd > 100\"\n",
```
[the remark this answers is not in the document]
Comment (vishwarajanand) [reply docs/vector_store.ipynb:667]: Added in the comment on line 665

File: docs/vector_store.ipynb [hunk 4]
```diff
@@ -563,6 +560,115 @@
     "\n",
     "print(docs)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a Vector Store using existing table\n",
```
[the remark this answers is not in the document]
Comment (vishwarajanand) [reply docs/vector_store.ipynb:568]: Adding the case of json filtering

File: docs/vector_store.ipynb [hunk 5]
```diff
@@ -563,6 +560,115 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set an existing table name\n",
+    "TABLE_NAME = \"products\"\n",
+    "\n",
+    "# Initialize AlloyDBVectorStore\n",
```
[the remark this answers is not in the document]
Comment (vishwarajanand) [reply docs/vector_store.ipynb:624]: Fixed it

File: docs/vector_store.ipynb [hunk 6]
```diff
@@ -559,7 +556,149 @@
...
+    "        product_id SERIAL PRIMARY KEY,\n",
+    "        name VARCHAR(255) NOT NULL,\n",
+    "        description TEXT,\n",
+    "        price_usd DECIMAL(10, 2) NOT NULL,\n",
+    "        category VARCHAR(255),\n",
+    "        quantity INT DEFAULT 0,\n",
+    "        sku VARCHAR(255) UNIQUE NOT NULL,\n",
+    "        image_url VARCHAR(255),\n",
+    "        metadata JSON,\n",
+    "        embed vector(768) DEFAULT NULL --> vector dimensions depends on the embedding model\n",
```
Comment (vishwarajanand) [inline docs/vector_store.ipynb:589]: I can make the dimensions lesser to make it look neater.

`textembedding-gecko@latest` produces a 768 size embedding, ignoring the dimensions param.

To reduce the vector size, a new embedding service would be needed in the notebook which might create some confusion:

```
from langchain_google_vertexai import VertexAIEmbeddings

embedding = VertexAIEmbeddings(
    model_name="text-embedding-004", project=PROJECT_ID
)
# following produces a 2 size embedding vector
embedding.embed(texts = ["sample"], dimensions = 2) // gecko ignores dimensions
```

So, for brevity, table creations schema is in a collapsed section by default and we are using a 768 size embedding

**Review by averikitsch — APPROVED** [review]

**Pull request merged**